reconcile/ldap_groups/integration.py

```diff
@@ -199,44 +172,57 @@ def get_desired_groups_for_roles(
                 contact_list=contact_list,
             )
             for role in roles
-            # roles with empty ldap_group are already filtered out in get_roles; just make mypy happy
             if role.ldap_group
         ]
 
-    def get_desired_groups_for_aws_groups(
+    def _filter_aws_account(
+        self, accounts: Iterable[AWSAccountSSO]
+    ) -> list[AWSAccountSSO]:
+        """Return a unique list of AWS accounts to be used for AWS roles."""
+        filtered_account = {}
+        for account in accounts:
+            if not account.sso:
+                continue
+            if not integration_is_enabled(self.name, account):
+                continue
+            if account.uid in filtered_account:
+                continue
+            filtered_account[account.uid] = account
+        return list(filtered_account.values())
```

Comment:
this filter logic is very similar to https://github.com/app-sre/qontract-reconcile/pull/4184/files#diff-f7cabfa1e5dd313115cdbc2a19fc224bd8754579c1d7863901dc870ad5039553R151-R166, should it be a common filter?